### PR TITLE
feat(state-sync): Pass sync adapter to network

### PR DESCRIPTION
**Pull request by VanBarbascu**
Pass message to state sync actor and client actor. 
Once we move the request logic to sync actor, we can stop sending the response to the client actor.

### CODE REVIEW

File: chain/client/src/sync/adapter.rs [hunk 1]
```diff
@@ -108,9 +109,14 @@ impl SyncAdapter {
 }
 
 /// Interface for network
-#[async_trait::async_trait]
 impl NetworkStateSync for SyncAdapter {
-    async fn send(&self, shard_uid: ShardUId, msg: StateSyncResponse) {
-        self.send(shard_uid, msg.with_span_context());
+    fn send(&self, info: StateResponseInfo) {
+        if let Some(&shard_uid) =
+            self.actor_handler_map.keys().find(|shard_uid| shard_uid.shard_id() == info.shard_id())
```
Comment (nikurt) [inline chain/client/src/sync/adapter.rs:115]: Oh, that won't scale well to tracking 1M shards. :)
Comment which assumptions make O(N) accesses acceptable.

File: chain/network/src/peer/peer_actor.rs [hunk 2]
```diff
@@ -1059,8 +1059,9 @@ impl PeerActor {
                     .await?
                     .map(PeerMessage::VersionedStateResponse),
                 PeerMessage::VersionedStateResponse(info) => {
-                        //TODO: Route to state sync actor.
-                    network_state.client.state_response(info).await;
+                    //TODO: Route to state sync actor.
```
Comment (nikurt) [inline chain/network/src/peer/peer_actor.rs:1062]:
```suggestion
                    // TODO: Route to state sync actor.
```

**Review by nikurt — APPROVED** [review]

**Review by marcelo-gonzalez — COMMENTED** [review]
Looks okay to me overall here, so feel free to merge. But in general I think I would caution against adding code that doesn't do anything yet. Because who knows if that's how you'll really want to do things once you hash out all the details later? I just think seeing unused code in core places (like here where we're sending a copy of the `StateResponse` message to both the `ClientActor` and the `SyncActor`, or storing an `Arc<RwLock<dyn state_sync::StateSync>>` in the `PeerManagerActor` when it doesn't actually do anything yet) adds a bit of overhead in just reading/understanding the code. Like, why not just add all this stuff when it's ready to do something?

Again, feel free to merge it. But I think it would make sense in the future to just make these changes as they appear as part of functional changes

**Comment by VanBarbascu** [comment]
I agree with you that having code that does not execute anything yet is not the way to go here. We can leave this PR open until it's time comes. We can bridge the gap between network and state sync when the we get there.

**Comment by VanBarbascu** [comment]
We are taking a different approach to sending state parts between peers. Will close this PR for now and come back to it at a later time when I have bandwidth for this refactoring.

**Pull request closed without merging**